scripts/scr_start_load/scr_start_load.gml

```diff
@@ -3,31 +3,42 @@ function scr_start_load(fleet, load_from_star, load_options) {
     // load_from_star: star object
 
     // this distributes the marines and vehicles to the correct ships if the chapter is fleet-based or a home-based chapter
-	var vet_count, scout_count, _unit, _comp, vet_list, vet_check,scout_list, scout_check, _marine;
-	var vet_role = obj_ini.role[100,3];
-	var scout_role = obj_ini.role[100,12];
+	var  _unit, _comp, _marine, total_vehic_size;
+	var total_distribute_squads = [[],[],[],[],[],[],[],[]];
 	var escort_load = load_options[0];
 	var split_scouts = load_options[1];
 	var split_vets = load_options[2];
-	if (split_vets == 1){
-		vet_list = [];
-		_comp = 1;
-		for (_unit =1; _unit<(array_length(obj_ini.role[_comp])-1); _unit++){
-			if (obj_ini.role[_comp,_unit] == vet_role){
-				array_push(vet_list, obj_ini.TTRPG[_comp,_unit]);
+	var comp_has_units=[];
+	for (_comp=0;_comp<10;_comp++;){
```

Comment:
as soon as we have proper Company structs, we should move all these loops into the Company structs. It's just way too much repetition.